pkg/controller/identity/iamaccesskey/controller.go

```diff
@@ -0,0 +1,167 @@
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.IAMAccessKey{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.IAMAccessKeyGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: iam.NewAccessClient}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(config aws.Config) iam.AccessClient
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cfg, err := awscommon.GetConfig(ctx, c.kube, mg, awscommon.GlobalRegion)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: c.newClientFn(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	client iam.AccessClient
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	keys, err := e.client.ListAccessKeysRequest(&awsiam.ListAccessKeysInput{UserName: aws.String(cr.Spec.ForProvider.IAMUsername)}).Send(ctx)
+	if resource.Ignore(iam.IsErrorNotFound, err) != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errList)
+	}
+
+	if keys != nil {
+		for _, key := range keys.AccessKeyMetadata {
+			if aws.StringValue(key.AccessKeyId) == cr.Status.AtProvider.AccessKeyID {
+				cr.SetConditions(runtimev1alpha1.Available())
+
+				return managed.ExternalObservation{
+					ResourceExists:   true,
+					ResourceUpToDate: string(key.Status) == cr.Status.AtProvider.Status,
+				}, nil
+			}
+		}
+	}
+
+	return managed.ExternalObservation{}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.CreateAccessKeyRequest(&awsiam.CreateAccessKeyInput{UserName: aws.String(cr.Spec.ForProvider.IAMUsername)}).Send(ctx)
+
+	var conn managed.ConnectionDetails = nil
+	if response != nil && response.AccessKey != nil {
+		conn = managed.ConnectionDetails{}
+		conn[runtimev1alpha1.ResourceCredentialsSecretUserKey] = []byte(aws.StringValue(response.AccessKey.AccessKeyId))
+		conn[runtimev1alpha1.ResourceCredentialsSecretPasswordKey] = []byte(aws.StringValue(response.AccessKey.SecretAccessKey))
+
+		cr.Status.AtProvider.Status = string(response.AccessKey.Status)
+		cr.Status.AtProvider.AccessKeyID = aws.StringValue(response.AccessKey.AccessKeyId)
```

Comment:
I think we need to use external-name annotation for access key id. See VPC controller for handling provider-assigned name.

Reply:
Ah, so it shouldn't be inside of the status?

Reply:
Ok all done!